src/coreclr/src/jit/hwintrinsicarm64.cpp

```diff
@@ -396,9 +396,8 @@ GenTree* Compiler::impSpecialIntrinsic(NamedIntrinsic        intrinsic,
 
                 for (unsigned i = 0; i < sig->numArgs; i++)
                 {
-                    tmp        = gtNewArgList(impPopStack().val);
-                    tmp->gtOp2 = op1;
-                    op1        = tmp;
+                    tmp = new (this, GT_LIST) GenTreeArgList(impPopStack().val, tmp);
```

Comment:
Should there be a `gtNewArgList` overload that does this instead? (I'm not clear on when a helper method vs a direct `new (this, genTreeOps)` should be used)

From a quick grep, are there any other similar codepaths that might be missing flag propagation?

Reply:
@tannergooding there are `gtNewArgList` overloads but those wrap arguments to list, e.g. https://github.com/dotnet/runtime/blob/master/src/coreclr/src/jit/gentree.cpp#L6475 (note that `gtNewArgList` for `argX`)

Reply:
Ah, actually there is `Compiler::gtNewListNode(GenTree* op1, GenTreeArgList* op2)` 🤔

Reply:
>  are there any other similar codepaths that might be missing flag propagation?

I couldn't find any